src/model/CoilCoolingDXCurveFitPerformance.hpp

```diff
@@ -0,0 +1,169 @@
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_COILCOOLINGDXCURVEFITPERFORMANCE_HPP
+#define MODEL_COILCOOLINGDXCURVEFITPERFORMANCE_HPP
+
+#include <model/ModelAPI.hpp>
+#include "ResourceObject.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+class Schedule;
+class CoilCoolingDX;
+class CoilCoolingDXCurveFitOperatingMode;
+
+namespace detail {
+
+  class CoilCoolingDXCurveFitPerformance_Impl;
+
+} // detail
+
+/** CoilCoolingDXCurveFitPerformance is a ResourceObject that wraps the OpenStudio IDD object 'OS:Coil:Cooling:DX:CurveFit:Performance'. */
+class MODEL_API CoilCoolingDXCurveFitPerformance : public ResourceObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit CoilCoolingDXCurveFitPerformance(const Model& model,
+                                            const CoilCoolingDXCurveFitOperatingMode& baseOperatingMode);
+
+  explicit CoilCoolingDXCurveFitPerformance(const Model& model,
+                                            const CoilCoolingDXCurveFitOperatingMode& baseOperatingMode,
+                                            const CoilCoolingDXCurveFitOperatingMode& alternativeOperatingMode1);
+
+  explicit CoilCoolingDXCurveFitPerformance(const Model& model,
+                                            const CoilCoolingDXCurveFitOperatingMode& baseOperatingMode,
+                                            const CoilCoolingDXCurveFitOperatingMode& alternativeOperatingMode1,
+                                            const CoilCoolingDXCurveFitOperatingMode& alternativeOperatingMode2);
+
+  virtual ~CoilCoolingDXCurveFitPerformance() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> capacityControlMethodValues();
+
+  static std::vector<std::string> compressorFuelTypeValues();
```

Comment:
:+1:

Reply:
(I'd also throw in `validXXXValues` just because the API is inconsistent throughout)